lib/puppet/type/archive.rb

```diff
@@ -7,9 +7,9 @@
   @doc = 'Manage archive file download, extraction, and cleanup.'
 
   ensurable do
-    desc 'whether archive file should be present/absent (default: present)'
+    desc 'whether archive file should be present/latest/absent (default: present)'
 
-    newvalue(:present) do
+    newvalues(:present, :latest) do
```

Comment:
Do not add latest to ensurable method. This module logic for updating the files is simply:

1. is ensure == present
2. does checksum parameter match expectation?

This is much simpler then using latest like most puppet modules, and provides a much cleaner way to let you know the file has updated. If you have a newer file, it simply lets you know the checksum has changed.